File: libgdx_drag_race/core/src/com/nopalsoft/dragracer/shop/NoAdsSubMenu.java
package com.nopalsoft.dragracer.shop;

import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
import com.nopalsoft.dragracer.Assets;
import com.nopalsoft.dragracer.MainStreet;
import com.nopalsoft.dragracer.Settings;

public class NoAdsSubMenu {

    int priceNoAds = 20000;

    TextButton buttonNoAds;
    Label labelNoAds;

    Table containerTable;
    MainStreet game;

    public NoAdsSubMenu(final MainStreet game, Table containerTable) {
        this.game = game;
        this.containerTable = containerTable;
        containerTable.clear();

        if (!Settings.didBuyNoAds)
            labelNoAds = new Label(priceNoAds + "", Assets.labelStyleSmall);

        buttonNoAds = new TextButton("Buy", Assets.styleTextButtonBuy);
        if (Settings.didBuyNoAds)
            buttonNoAds.setVisible(false);
        addPressEffect(buttonNoAds);
        buttonNoAds.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                if (Settings.coinsTotal >= priceNoAds) {
                    Settings.coinsTotal -= priceNoAds;
                    Settings.didBuyNoAds = true;
                    labelNoAds.setVisible(false);
                    buttonNoAds.setVisible(false);
                }
            }
        });

        // Upgrade BoostTime
        containerTable.add(new Image(Assets.horizontalSeparatorDrawable)).expandX().fill()
                .height(5);
        containerTable.row();
        containerTable
                .add(addPlayerTable(labelNoAds,
                        Assets.buttonNoAds,
                        buttonNoAds))
                .expandX().fill();
        containerTable.row();
    }

    private Table addPlayerTable(Label labelPrice,
                                 TextureRegionDrawable imageDrawable, TextButton button) {

        Image coinImage = new Image(Assets.coinFront);
        Image playerImage = new Image(imageDrawable);

        if (labelPrice == null)
            coinImage.setVisible(false);

        Table titleBarTable = new Table();
        titleBarTable.add(new Label("No more ads", Assets.labelStyleSmall)).expandX()
                .left().padLeft(5);
        titleBarTable.add(coinImage).right();
        titleBarTable.add(labelPrice).right().padRight(10);

        Table descriptionTable = new Table();
        descriptionTable.add(playerImage).left().pad(10).size(55, 45);
        Label descriptionLabel = new Label("Buy it and no more ads will appear in the app", Assets.labelStyleSmall);
        descriptionLabel.setWrap(true);
        descriptionLabel.setFontScale(.85f);
        descriptionTable.add(descriptionLabel).expand().fill().padLeft(5);

        Table tbContent = new Table();
        tbContent.add(titleBarTable).expandX().fill().colspan(2).padTop(8);
        tbContent.row().colspan(2);
        tbContent.add(descriptionTable).expandX().fill();
        tbContent.row().colspan(2);

        tbContent.add(button).right().padRight(10).size(120, 45);

        tbContent.row().colspan(2);
        tbContent.add(new Image(Assets.horizontalSeparatorDrawable)).expandX().fill()
                .height(5).padTop(15);

        return tbContent;
    }

    protected void addPressEffect(final Actor actor) {
        actor.addListener(new InputListener() {
            @Override
            public boolean touchDown(InputEvent event, float x, float y,
                                     int pointer, int button) {
                actor.setPosition(actor.getX(), actor.getY() - 3);
                event.stop();
                return true;
            }

            @Override
            public void touchUp(InputEvent event, float x, float y,
                                int pointer, int button) {
                actor.setPosition(actor.getX(), actor.getY() + 3);
            }
        });
    }
}

File: libgdx_drag_race/core/src/com/nopalsoft/dragracer/shop/NoAdsSubMenu.kt
package com.nopalsoft.dragracer.shop

import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.ui.TextButton
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable
import com.nopalsoft.dragracer.Assets
import com.nopalsoft.dragracer.MainStreet
import com.nopalsoft.dragracer.Settings

open class NoAdsSubMenu(var game: MainStreet, container: Table) {
    var priceNoAds: Int = 20000

    var buttonNoAds: TextButton
    var labelNoAds: Label? = null

    init {
        container.clear()

        if (!Settings.didBuyNoAds) labelNoAds =
            Label(priceNoAds.toString() + "", Assets.labelStyleSmall)

        buttonNoAds = TextButton("Buy", Assets.styleTextButtonBuy)
        if (Settings.didBuyNoAds) buttonNoAds.isVisible = false
        addPressEffect(buttonNoAds)
        buttonNoAds.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                if (Settings.coinsTotal >= priceNoAds) {
                    Settings.coinsTotal -= priceNoAds
                    Settings.didBuyNoAds = true
                    labelNoAds!!.isVisible = false
                    buttonNoAds.isVisible = false
                }
            }
        })

        // Upgrade boost time
        container.add(Image(Assets.horizontalSeparator)).expandX().fill().height(5f)
        container.row()
        container.add(
            addCharacterTable(
                labelNoAds,
                Assets.buttonNoAds,
                buttonNoAds
            )
        )
            .expandX()
            .fill()
        container.row()
    }

    private fun addCharacterTable(
        labelPrice: Label?,
        images: TextureRegionDrawable,
        button: TextButton
    ): Table {
        val imageCoinFront = Image(Assets.coinFront)
        val imageCharacter = Image(images)

        if (labelPrice == null) imageCoinFront.isVisible = false

        val tableTitleBar = Table()
        tableTitleBar.add(Label("No more ads", Assets.labelStyleSmall)).expandX()
            .left().padLeft(5f)
        tableTitleBar.add(imageCoinFront).right()
        tableTitleBar.add(labelPrice).right().padRight(10f)

        val tableDescription = Table()
        tableDescription.add(imageCharacter).left().pad(10f).size(55f, 45f)
        val labelDescription =
            Label("Buy it and no more ads will appear in the app", Assets.labelStyleSmall)
        labelDescription.wrap = true
        labelDescription.setFontScale(.85f)
        tableDescription.add(labelDescription).expand().fill().padLeft(5f)

        val tableContent = Table()
        tableContent.add(tableTitleBar).expandX().fill().colspan(2).padTop(8f)
        tableContent.row().colspan(2)
        tableContent.add(tableDescription).expandX().fill()
        tableContent.row().colspan(2)

        tableContent.add(button).right().padRight(10f).size(120f, 45f)

        tableContent.row().colspan(2)
        tableContent.add(Image(Assets.horizontalSeparator)).expandX().fill()
            .height(5f).padTop(15f)

        return tableContent
    }

    private fun addPressEffect(actor: Actor) {
        actor.addListener(object : InputListener() {
            override fun touchDown(
                event: InputEvent, x: Float, y: Float,
                pointer: Int, button: Int
            ): Boolean {
                actor.setPosition(actor.x, actor.y - 3)
                event.stop()
                return true
            }

            override fun touchUp(
                event: InputEvent, x: Float, y: Float,
                pointer: Int, button: Int
            ) {
                actor.setPosition(actor.x, actor.y + 3)
            }
        })
    }
}
